### DISCUSSION TITLE
[과목 2] 제 3 관리 구문

### DISCUSSION THREAD

**Discussion by white-gyu** [flataex/sqlp-study#18] [Custom 연습문제]
문제 1. 주어진 두 개의 테이블에 대해서 아래의 SQL문을 수행한 이후에 TEST1 테이블의 건수는?

```oracle
[TEST1]
COL1  COL2  COL3
------------------
A      X      1
B      Y      2
C      Z      3

[TEST2]
COL1  COL2  COL3
------------------
A      X      1
B      Y      2
C      Z      3
D      가      4
E      나      5

[SQL]
MERGE INTO TEST1
USING TEST2
 ON (TEST1.COL1 = TEST2.COL1)
WHEN MATCHED THEN
 UPDATE SET TEST1.COL3 = 4
      WHERE TEST1.COL3 = 2
 DELETE WHERE TEST1.COL3 <= 2
WHEN NOT MATCHED THEN
 INSERT(TEST1.COL1, TEST1.COL2, TEST1.COL3)
 VALUES(TEST2.COL1, TEST2.COL2, TEST2.COL3);
```

1) 4     
2) 3    
3) 5
4) 8

---

문제 2. 다음 보기 중 데이터베이스 테이블의 제약조건(Constraint)에 대한 설명으로 올바르지 않은 것은?

1) 외래키(Foreign Key)는 두 개의 테이블 간의 참조 무결성을 제약한다.
2) 기본키(Primary Key) 제약사항은 테이블 당 하나만 제약할 수 있다.
3) Check 제약조건(Constraint)은 특정값만 입력되게 제약한다.
4) 고유키(Unique Key) 제약이 설정 되면 NULL값을 가질 수 없다.

---

문제 3. 아래의 상품 마스터 테이블에 대한 상품ID '001' 의 최종 상품명은 무엇인가?

```oracle
[테이블 : 상품마스터]
상품ID    상품명
-------------
001      TV

[SQL]
BEGIN TRANSACTION;
SAVE TRANSACTION SP1;
UPDATE 상품마스터 SET 상품명 = 'LG-TV' WHERE 상품ID = '001';
SAVE TRANSACTION SP2;
UPDATE 상품마스터 SET 상품명 = '평면-TV' WHERE 상품ID = '001';
ROLLBACK TRANSACTION SP2;
COMMIT;
```

**Comment by Psjepark** [comment]
> 문제 1. 주어진 두 개의 테이블에 대해서 아래의 SQL문을 수행한 이후에 TEST1 테이블의 건수는?
> 
> ```
> [TEST1]
> COL1  COL2  COL3
> ------------------
> A      X      1
> B      Y      2
> C      Z      3
> 
> [TEST2]
> COL1  COL2  COL3
> ------------------
> A      X      1
> B      Y      2
> C      Z      3
> D      가      4
> E      나      5
> 
> [SQL]
> MERGE INTO TEST1
> USING TEST2
>  ON (TEST1.COL1 = TEST2.COL1)
> WHEN MATCHED THEN
>  UPDATE SET TEST1.COL3 = 4
>       WHERE TEST1.COL3 = 2
>  DELETE WHERE TEST1.COL3 <= 2
> WHEN NOT MATCHED THEN
>  INSERT(TEST1.COL1, TEST1.COL2, TEST1.COL3)
>  VALUES(TEST2.COL1, TEST2.COL2, TEST2.COL3);
> ```
> 
> 

- [x] 1. 4

- [ ] 2. 3

- [ ] 3. 5

- [ ] 4. 8

> 
> 문제 2. 다음 보기 중 데이터베이스 테이블의 제약조건(Constraint)에 대한 설명으로 올바르지 않은 것은?
> 
> - [ ] 1. 외래키(Foreign Key)는 두 개의 테이블 간의 참조 무결성을 제약한다.
> - [ ] 2. 기본키(Primary Key) 제약사항은 테이블 당 하나만 제약할 수 있다.
> - [ ] 3. Check 제약조건(Constraint)은 특정값만 입력되게 제약한다.
> - [x] 4. 고유키(Unique Key) 제약이 설정 되면 NULL값을 가질 수 없다.
> 
> 문제 3. 아래의 상품 마스터 테이블에 대한 상품ID '001' 의 최종 상품명은 무엇인가?
> 
> ```
> [테이블 : 상품마스터]
> 상품ID    상품명
> -------------
> 001      TV
> 
> [SQL]
> BEGIN TRANSACTION;
> SAVE TRANSACTION SP1;
> UPDATE 상품마스터 SET 상품명 = 'LG-TV' WHERE 상품ID = '001';
> SAVE TRANSACTION SP2;
> UPDATE 상품마스터 SET 상품명 = '평면-TV' WHERE 상품ID = '001';
> ROLLBACK TRANSACTION SP2;
> COMMIT;
> ```
> LG-TV

**Comment by beolggoolosori** [comment]
1. `1`
2. `4`
3. LG-TV
